### ISSUE TITLE
NFDataList::Clear() did not reset type when clear data

### ISSUE THREAD

**Issue by zhxilin** [ketoo/NoahGameFrame#63]
`NFDataList::Clear()` has 2 bugs:
1. never reset type for the item that has been removed
```
NFDataList xDataList;
for (int i = 0; i < 10; ++i)
{
	xDataList << i;
}
xDataList.Clear();
xDataList << "Hello NF"; //This line will not succeed
```
After clear the NFDataList, append new data of different type will fail.

2. first `STACK_SIZE` items will not be removed
```
NFDataList xDataList;
xDataList << 10;
xDataList.Clear(); 
```
It will only set `mnUseSize` to 0 without reset data, cause `mvList.size() < STACK_SIZE`

**Comment by zhxilin** [comment]
I've post a fix, see 254a92b14375a54abc3ddc76fb21ffc0c86b399a

**Comment by ketoo** [comment]
Thanks, I have merged your pull request. Good job.
